lib/scoped/index.js

```diff
@@ -32,8 +32,16 @@ function scoped (api, type) {
     }
   }
 
-  api.on('change', require('../utils/handle-type-change').bind(null, emitter, type))
+  // possible race condition... if for some reason this didn't resolve before
+  // a remote change got pulled in, add/remove/change events wouldn't fire.
+  // seems unlikely, but, still...
+  scopedApi.findAll().then(function (objects) {
+    objects.forEach(function (object) {
+      state.objectTypeById[object.id] = object.type
+    })
+  })
```

Comment:
I don’t think this needs to be run every time we initialise a scoped API, because `state.objectTypeById` is shared globally. I’d run this only once in the constructor. 

To avoid the race condition, we can add a setup promise, that needs to be fulfilled before running any APIs. But let’s do that in a follow up issue, one challenge at a time :)